### PR TITLE
Fix default k8s secrets path

### PR DESCRIPTION
**Pull request by azdagron**
The official path to these files is under /var/run/ not /run. We've only managed to get by using /run because the two are symlink'd in many distros. However, there is no such symlink in scratch environments. We should be using the official path anyway.

I surmise the risk here is low, except maybe in some one off test environments. The official paths should be available in any production K8s system.

### CODE REVIEW

**Comment by evan2645** [comment]
It was my impression that `/var/run` was being deprecated in favor of `/run` and that the symlink is the compat layer. Is it possible that there are distros out there which don't symlink?

**Comment by azdagron** [comment]
`/run` is the proper path. However, all kubernetes docs reference `/var/run`....

@rturner3 suggested maybe we just create the symlink in our scratch image. That actually sounds ok to me instead of all this. We have other places where we default to `/var/run` though (psat token default path)...

**Comment by azdagron** [comment]
I'm going to close this, and for now we can create the symlink in our scratch image. We can consider long term what we want to do.

**Comment by azdagron** [comment]
RESURRECTION! Reversing course. This seems like the right way forward since Kubernetes is strongly "/var/run" and there are concerns about "/run" and "/var/run" diverging under strange circumstances.

**Comment by evan2645** [comment]
From a separate conversation, @azdagron has said:
>  Kubernetes is mounting under /var/run. They are creating the directory if it doesn't exist, which means if the symlink doesn't exist, /var/run and /run will diverge.

It seems like a good idea to reference the path that k8s is mounting, even if the new hotness is elsewhere. I suspect they mount /var/run for exactly the reason we're running into here. In light of above, I'm supportive of adopting /var/run over /run

File: test/integration/suites/k8s-scratch/conf/agent/spire-agent.yaml [hunk 1]
```diff
@@ -73,7 +73,6 @@ data:
           # Minikube does not have a cert in the cluster CA bundle that
           # can authenticate the kubelet cert, so skip validation.
           skip_kubelet_verification = true
-          token_path = "/var/run/secrets/kubernetes.io/serviceaccount/token"
```
Comment (evan2645) [inline test/integration/suites/k8s-scratch/conf/agent/spire-agent.yaml:76]: the evidence, which i totally missed when the test suite came in

**Review by evan2645 — APPROVED** [review]
LGTM

I know @rturner3 was also interested

**Pull request merged**